tests/EStimLibrary.UnitTests/Core/HardwareInterfaces/NeuralInterfaceManagerTests.cs

```diff
@@ -0,0 +1,207 @@
+        Assert.Equal(interfaceType, neuralInterface.GetType());
+
+        // That the new neural interface object is created with the given object[] parameters.
+        if (interfaceType == typeof(ContactGroup) && neuralInterface is ContactGroup contactGroup) {
+            Assert.Equal(interfaceSpecificParams[0], contactGroup.NumContacts); 
+        }
+    }
+
+    /// <summary>
+    /// The new neural interface object of the correct Type is created successfully with the given object[] parameters.
+    /// Test with classes that requires no additional parameters.
+    /// </summary>
+    /// <param name="interfaceType">The type of the hardware interface, for example: ContactGroup, GelPad</param>
+    /// <param name="interfaceSpecificParams">Parameters specific to that interface</param>
+    [Theory]
+    [InlineData(typeof(GelPad), new object[]{})]
+    public void CreateAndRegisterNeuralInterface_ShouldInitWithCorrectParams_WhenValidType_NoAdditionalParams(Type interfaceType, object[] interfaceSpecificParams) 
+    {
+        // Arrange
+        var NIManager = new NeuralInterfaceManager();
+
+        // Act
+        var result = NIManager.CreateAndRegisterNeuralInterface(interfaceType, interfaceSpecificParams, out int globalInterfaceId);
+        NIManager.TryGetNeuralInterface(globalInterfaceId, out NeuralInterfaceHardware neuralInterface);
+
+        // Assert
+        // That the new neural interface object is successfully created
+        Assert.NotNull(neuralInterface);
+
+        // That the new neural interface object is of the correct Type
+        Assert.Equal(interfaceType, neuralInterface.GetType());
+
+        // That the new neural interface object is created with the given object[] parameters, in this case, none.
+        if (interfaceType == typeof(GelPad) && neuralInterface is GelPad gelPad) {
+            Assert.Equal(new GelPad().NumContacts, gelPad.NumContacts); 
+        }
+    }
+
+    /// <summary>
+    /// The manager assigns global contact IDs for the new neural interface. 
+    /// Test that this ID is returned by the method CreateAndRegisterNeuralInterface.
+    /// </summary>
+    /// <param name="interfaceType">The type of the hardware interface, for example: ContactGroup, GelPad</param>
+    /// <param name="interfaceSpecificParams">Parameters specific to that interface</param>
+    [Theory]
+    [InlineData(typeof(ContactGroup), new object[]{1})]
+    [InlineData(typeof(GelPad), new object[]{})]
+    public void CreateAndRegisterNeuralInterface_ShouldReturnAssignedGlobalContactIDs(Type interfaceType, object[] interfaceSpecificParams)
+    {
+        // Arrange
+        var NIManager = new NeuralInterfaceManager();
+
+        // Act
+        var contactIds = NIManager.CreateAndRegisterNeuralInterface(interfaceType, interfaceSpecificParams, out _);
+
+        // Assert
+        Assert.NotEmpty(contactIds);
+        Assert.All(contactIds, id => Assert.True(NIManager.IsValidContactId(id)));
```

Comment:
These are good checks! Similar to comment above, can also pass in the explicit expected set of contact ids and make sure that matches too